doc/viper/viper.md

```diff
@@ -0,0 +1,148 @@
+# Vitess Viper Guidelines
+
+### Preface
+
+## What is Viper?
+
+[`viper`][viper] is a configuration-management library for Go programs.
+It acts as a registry for configuration values coming from a variety of sources, including:
+
+- Default values.
+- Configuration files (JSON, YAML, TOML, and other formats supported), including optionally watching and live-reloading.
+- Environment variables.
+- Command-line flags, primarily from `pflag.Flag` types.
+
+It is used by a wide variety of Go projects, including [hugo][hugo] and [the kubernetes operator][kops].
+
+## Common Usage
+
+Broadly speaking, there's two approaches for Vitess to using viper.
+
+### Approach 1: Everything in the global registry.
```

Comment:
@mattlord absolutely not urgent, but curious if you have any strong reactions (positive or negative!) to the 2(and a half) approaches outlined here

I'm gonna noodle on this in the background but broadly turn my attention today to working out how to best use config files (also, if you have any good examples of configs that we would want to be dynamic (query timeouts, and other "threshold-y" type things, probably?) let me know!